src/utilities/sign.jl

```diff
@@ -63,20 +63,18 @@ function sign_aws4!(aws::AbstractAWSConfig, request::Request, time::DateTime)
     # Authentication scope string...
     authentication_scope = join(authentication_scope, "/")
 
-    # SHA256 hash of content...
-    content_hash = bytes2hex(digest(MD_SHA256, request.content))
+    # SHA256 hash of content if data is in memory
+    if !haskey(request.headers, "x-amz-content-sha256")
+        content_hash = bytes2hex(digest(MD_SHA256, request.content))
+        request.headers["x-amz-content-sha256"] = content_hash
+    else
+        content_hash = "UNSIGNED-PAYLOAD"
+    end
```

Comment:
You can write this a bit more concisely like so:
```suggestion
    content_hash = get!(request.headers, "x-amz-content-sha256") do
        bytes2hex(digest(MD_SHA256, request.content))
    end
```